### PR TITLE
Moving to TypeScript

### PR DESCRIPTION
**Pull request by pvinis**
# Summary

We should finally move this to typescript. Here is the first pass.

I added typescript and the react and react native types, I made a base config, I added `files` in package.json, I renamed all files, and I fixed the errors.

To compile we do `yarn build`, and this should also happen before `npm publish`, we can add it in a `prepublish`.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅❌     |
| Android |    ✅❌     |

## Checklist

- [ ] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

**Comment by pvinis** [comment]
This is ready to be reviewed! I fixed a bunch of types, and now you can do `yarn build` and `yarn release` and everything should work fine.

File: src/useKeyboard.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  useEffect(() => {
+    const keyboardWillShowListener = Keyboard.addListener('keyboardWillShow', handleKeyboardWillShow)
+    const keyboardDidShowListener = Keyboard.addListener('keyboardDidShow', handleKeyboardDidShow)
+    const keyboardWillHideListener = Keyboard.addListener('keyboardWillHide', handleKeyboardWillHide)
+    const keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', handleKeyboardDidHide)
+
+    return () => {
+		keyboardWillShowListener.remove()
```
Comment (melihberberolu) [inline src/useKeyboard.ts]: Can you fix this reformat ?

**Comment by melihberberolu** [comment]
@dabit3 Can you check this PR ? I think everything is fine.

**Comment by pvinis** [comment]
I didn't bother formatting because I saw a PR adding eslint, but yes, I can run a prettier in this or something.

**Comment by pvinis** [comment]
Should I throw a prettierrc in here, or just run it?

**Comment by pvinis** [comment]
Nah, I will not add prettier or eslint here, I just run it and it's good for now. We should make a different PR for adding these tools.

**Comment by pvinis** [comment]
Bump! This should be ready to review again and merge if approved.

**Pull request merged**